src/bases.jl

```diff
@@ -736,29 +736,30 @@ function gram_schmidt(
     Ξ = empty(V)
     dim = manifold_dimension(M)
     N < dim && @warn "Input only has $(N) vectors, but manifold dimension is $(dim)."
+    linear_independent = true
     @inbounds for n in 1:N
-        Ξₙ = V[n]
+        Ξₙ = copy(M, p, V[n])
         for k in 1:length(Ξ)
             Ξₙ .-= real(inner(M, p, Ξ[k], Ξₙ)) .* Ξ[k]
         end
         nrmΞₙ = norm(M, p, Ξₙ)
         if nrmΞₙ == 0
             warn_linearly_dependent && @warn "Input vector $(n) has length 0."
-            @goto skip
+            linear_independent = false
         end
         Ξₙ ./= nrmΞₙ
         for k in 1:length(Ξ)
-            if !isapprox(real(inner(M, p, Ξ[k], Ξₙ)), 0; kwargs...)
+            if isapprox(abs(real(inner(M, p, Ξ[k], Ξₙ))), 1; kwargs...)
```

Comment:
I even think that this test is not necessary. If it would be linearly dependent of the others, we would have gotten a zero vector already.

Reply:
I don't understand, why did you change this line? Could you give an example this fixes? This test is primarily aimed at detecting numerical errors. Do you think they are harmless? Naively it seems valuable to catch vectors that aren't exactly 0 (see that the test above is `== 0`, not approximately 0) but still too close to zero to be useful in practice.

Reply:
So we often lost a vector to this test, which was perfectly fine to an ONB (I can only run into this in the code of my student currently and do not have a separate example), but if the norm of \Xi_n is not zero and one of these tests fires (either mine our yours, I think mine is more stable) – it should really just be numerical errors and I do not see why that should lead to the conclusion that the vector should not be part of the final set.

Also I think that the copy I am now doing (of `V`) since Gram Schmidt is non mutating is causing the three new errors?

Reply:
One of the tests seems to explicitly check this condition. Could you maybe keep this check as an option that can be turned on through a keyword argument?

I totally agree that there should be a copy. The other two tests seem to fail because they assume that no copy is made.

Reply:
Thanks for checking.

I am ok with only turning this on by keyword, we do that for the return and the warning, too; but I find it strange to exclude vectors from the resulting basis just due to rounding errors?

Reply:
I don't exactly remember the motivation, but since it's already there, we can keep it just in case.

Reply:
Which one should we keep then?

The new one checks for linear dependence as the warning suggests

The old one – checked whether \Xi_n was not orthogonal to any of the vectors from the basis (but warned about linear dependence)...

Reply:
I'm not sure, does either one make the test pass?

Reply:
well the previous one did for sure, but then at least the warning was wrong and I am against throwing a vector out for numerical instability. We can also have both, but I do not see much of a use in the (old) orthogonality test if before that one subtracts all parts (or in other words projects on the orthogonal complement) – for sure its orthogonal then (upto numerical precision).

Reply:
OK, so let's just remove this test for now.

Reply:
Ok. Then I only have to understand what the failing tests were assumed to be testing and adapt or delete them

Reply:
Hm, so one of the examples in the tests now produces unit vectors where all but two are orthogonal, but these two — that is strange, are not.